python/fledge/plugins/north/azure_iot/azure_iot.py

```diff
@@ -21,34 +22,35 @@
 _LOGGER = logger.setup(__name__, level=logging.INFO)
 
 _DEFAULT_CONFIG = {
-    'plugin': {
-         'description': 'Azure North Plugin',
-         'type': 'string',
-         'default': 'azure_iot',
-         'readonly': 'true'
+    "plugin": {
+        "description": "Azure North Plugin",
+        "type": "string",
+        "default": "azure_iot",
+        "readonly": "true"
     },
-    'primaryConnectionString': {
-        'description': 'Connection string based on primary key used in API calls which allows device to '
-                       'communicate with Azure IoT Hub',
-        'type': 'string',
-        'default': 'HostName=<Host Name>;DeviceId=<Device Name>;SharedAccessKey=<Device Key>',
-        'order': '1',
-        'displayName': 'Primary Connection String'
+    "primaryConnectionString": {
+        "description": "Connection string based on primary key used in API calls which allows "
+                       "device to communicate with Azure IoT Hub",
+        "type": "string",
```

Comment:
We need to think of splitting this or make type password, as this has sensitive info.

Setting type password is straight-forward but returning *** turns into a challenge;

At some point, we need to handle logging of this and in support bundle stuff too.

Reply:
> We need to think of splitting this

There are different values format for connection string:

a) IoT Hub connection string
`HostName=<Host Name>;SharedAccessKeyName=<Key Name>;SharedAccessKey=<SAS Key>`
b) IoT Hub’s Event Hub-compatible connection string
`Endpoint=<ENDPOINT>;SharedAccessKeyName=<Key Name>;SharedAccessKey=<KEYVALUE>`
c) IoT Hub device connection string
`HostName=<Host Name>;DeviceId=<Device Name>;SharedAccessKey=<Device Key>`

If we really want to split the connection string config item with different individual items then we need to add
```
-> connection format - enumeration
-> Host Name - string
-> SharedAccessKeyName - string/password
-> SharedAccessKey - string/password
-> Endpoint - string
-> DeviceId - string

* with some validity optional attribute fto show/hide as per connection format
```

> Setting type password is straight-forward but returning *** turns into a challenge;

Absolutely right; 

> At some point, we need to handle logging of this and in support bundle stuff too.

This also seems a tricky/challenging one to hide this sensitive info from logs and support bundle.

Reply:
Let's log a Jira for this.